python/tests/api/types/test_type_util.py

```diff
@@ -0,0 +1,36 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import unittest
+
+from iceberg.api.types import (BooleanType, NestedField, StructType)
+from iceberg.api.types import type_util
+from iceberg.exceptions import ValidationException
+
+
+class TestConversions(unittest.TestCase):
+
+    def test_validate_schema_via_index_by_name(self):
```

Comment:
a few more test cases wouldn't be bad. A passing test case. Maybe an obvious wrong one (eg add `a` to the schema twice)

Reply:
Agree.  it would be great to have some failure cases.

Reply:
👌  will do

Reply:
@rymurr @TGooch44  updated the tests. Can you take another look? Thanks.